src/app/space/add-app-overlay/add-app-overlay.component.spec.ts

```diff
@@ -0,0 +1,164 @@
+        links: {
+        self: 'mock-self'
+        },
+        relationships: {
+        areas: { links: { related: 'mock-related' } },
+        iterations: { links: { related: 'mock-related' } },
+        workitemtypegroups: { links: { related: 'mock-related' } },
+        'owned-by': {
+            data: {
+            id: mockUser.id,
+            type: mockUser.type
+            }
+        }
+        },
+        relationalData: {
+        creator: mockUser
+        }
+    };
+
+    let mockSpaceTemplates: ProcessTemplate[] = [{
+        attributes: {
+        'can-construct': false,
+        description: 'Description-1',
+        name: 'Template - 01'
+        },
+        id: 'template-01',
+        type: 'spacetemplates'
+    }, {
+        attributes: {
+        'can-construct': true,
+        description: 'Description-2',
+        name: 'Template - 02'
+        },
+        id: 'template-02',
+        type: 'spacetemplates'
+    }, {
+        attributes: {
+        'can-construct': true,
+        description: 'Description-3',
+        name: 'Template - 03'
+        },
+        id: 'template-03',
+        type: 'spacetemplates'
+    }] as ProcessTemplate[];
+
+    let mockMessage: Notification = {
+        message: `Failed to create "${mockSpace.name}"`,
+        type: NotificationType.DANGER
+    };
+
+    beforeEach(async(() => {
+        TestBed.configureTestingModule({
+            imports: [
+                FormsModule,
+                HttpModule,
+                PopoverModule.forRoot()
+            ],
+            declarations: [
+                AddAppOverlayComponent
+            ],
+            providers: [
+                { provide: DependencyCheckService, useValue: mockDependencyCheckService },
+                PopoverConfig,
+                { provide: Broadcaster, useValue: mockBroadcaster },
+                { provide: Router, useValue: mockRouter },
+                { provide: SpaceTemplateService, useValue: mockSpaceTemplateService },
+                { provide: SpaceService, useValue: mockSpaceService },
+                { provide: Notifications, useValue: mockNotifications },
+                { provide: UserService, useValue: mockUserService },
+                { provide: SpaceNamespaceService, useValue: mockSpaceNamespaceService },
+                { provide: SpaceNamePipe, useValue: mockSpaceNamePipe },
+                { provide: SpacesService, useValue: mockSpacesService },
+                { provide: ContextService, useValue: mockContextService },
+                { provide: Logger, useValue: mockLogger },
+                { provide: ErrorHandler, useValue: mockErrorHandler }
+            ]
+        }).compileComponents();
+    }));
+
+    beforeEach(() => {
+        fixture = TestBed.createComponent(AddAppOverlayComponent);
+        component = fixture.componentInstance;
+        fixture.detectChanges();
+    });
+
+    it('should create', () => {
+        expect(component).toBeTruthy();
```

Comment:
How is this test fixing SEV2 / P1 issue? https://github.com/openshiftio/openshift.io/issues/3418 /cc @gorkem @joshuawilson